bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -158,7 +158,7 @@
     private SafeCaller safeCaller;
     private volatile boolean active = false;
     private StorageService storageService;
-    private Storage<Boolean> storage;
+    private Storage<String> storage;
```

Comment:
Because we don't really use the boolean value. When a thing gets enabled we simply remove the storage entry [see here](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java#L1220)

Reply:
@htreu Yes, the storage should handle it.
I assume it has been changed from boolean to string to not depend on some strange deserialization error (but I don't know).
Perhaps a "byte[]" and an static empty one would make most sense to perform (de)serialization as fast as possible.

Reply:
Boolean was removed because is is missleading. The storage will never return "true". It will return either false or will not contain the key at all.

Reply:
Agreed, marker is enough.